examples/fast_neural_style/neural_style.py

```diff
@@ -0,0 +1,291 @@
+    np.random.seed(seed)
+    torch.manual_seed(seed)
+
+
+def train(args):
+    device = torch.device("cuda" if args.cuda else "cpu")
+
+    transform = transforms.Compose([
+        transforms.Resize(args.image_size),
+        transforms.CenterCrop(args.image_size),
+        transforms.ToTensor(),
+        transforms.Lambda(lambda x: x.mul(255))
+    ])
+
+    if args.dataset == 'cifar10':
+        train_dataset = datasets.CIFAR10(root='/tmp/cifar10', train=True, download=True, transform=transform)
+    else:
+        train_dataset = datasets.ImageFolder(args.dataset, transform)
+
+    train_loader = DataLoader(train_dataset, batch_size=args.batch_size, shuffle=True, num_workers=0)
+
+    transformer = TransformerNet().to(device)
+    optimizer = Adam(transformer.parameters(), args.lr)
+    mse_loss = torch.nn.MSELoss()
+
+    vgg = Vgg16(requires_grad=False).to(device)
+    style_transform = transforms.Compose([
+        transforms.ToTensor(),
+        transforms.Lambda(lambda x: x.mul(255))
+    ])
+    if args.style_image != 'random':
+        style = utils.load_image(args.style_image, size=args.style_size)
+    else:
+        imarray = np.random.rand(256, 256, 3).astype('uint8')
+        style = Image.fromarray(imarray)
+
+    style = style_transform(style)
+
+    style = style.repeat(args.batch_size, 1, 1, 1).to(device)
+
+    features_style = vgg(utils.normalize_batch(style))
+    gram_style = [utils.gram_matrix(y) for y in features_style]
+
+    running_avgs = OrderedDict()
+    output_stream = sys.stdout
+    alpha = 0.98
+
+    def step(engine, batch):
+
+        x, _ = batch
+        x = x.to(device)
+
+        n_batch = len(x)
+
+        transformer.zero_grad()
+
+        y = transformer(x)
+
+        x = utils.normalize_batch(x)
+        y = utils.normalize_batch(y)
+
+        features_x = vgg(x)
+        features_y = vgg(y)
+
+        content_loss = args.content_weight * mse_loss(features_y.relu2_2, features_x.relu2_2)
+
+        style_loss = 0.
+        for ft_y, gm_s in zip(features_y, gram_style):
+            gm_y = utils.gram_matrix(ft_y)
+            style_loss += mse_loss(gm_y, gm_s[:n_batch, :, :])
+        style_loss *= args.style_weight
+
+        total_loss = content_loss + style_loss
+        total_loss.backward()
+        optimizer.step()
+
+        return {
+            'content_loss': content_loss.item(),
+            'style_loss': style_loss.item(),
+            'total_loss': total_loss.item()
+        }
+
+    trainer = Engine(step)
+    checkpoint_handler = ModelCheckpoint(args.checkpoint_model_dir, 'checkpoint',
+                                         save_interval=args.checkpoint_interval,
+                                         n_saved=10, require_empty=False, create_dir=True)
+
+    @trainer.on(Events.ITERATION_COMPLETED)
+    def update_logs(engine):
+        for k, v in engine.state.output.items():
+            old_v = running_avgs.get(k, v)
+            new_v = alpha * old_v + (1 - alpha) * v
```

Comment:
Can we use ignite metrics for that ?

Reply:
Yes, we can. I got that from the dcgan example. I'll look into that this weekend.

Reply:
@vfdev-5 Looking into this more, I'm not exactly sure how to incorporate Ignite Metrics into this. 

From what I have seen of the Metrics documentation, metrics are calculated using the output of the `step function` of the engine. In this case the metrics are the content_loss, style_loss and total_loss (MSE) that are needed for the optimization step inside the `step(engine, batch)` function. 

There are no metrics calculated at the end of Epoch, like in a classification problem.

This is why I used the `update_logs(engine)` function as shown. 

Happy to run with any ideas you may have.

Reply:
@anmolsjoshi I see your point, true that here there are running loss function values and we can not use Metrics in this case.